src/core/api/__tests__/public_api.test.ts

```diff
@@ -0,0 +1,571 @@
+      });
+    });
+
+    describe("getUrl", () => {
+      it("should return undefined in getUrl by default", () => {
+        expect(player.getUrl()).to.equal(undefined);
+      });
+    });
+
+    describe("getVideoDuration", () => {
+      /* tslint:disable:max-line-length */
+      it("should return the video element initial duration in getVideoDuration by default", () => {
+      /* tslint:enable:max-line-length */
+
+        // ! HAHA ! NaN is not === to NaN
+        const videoElement = player.getVideoElement();
+        if (videoElement == null) {
+          throw new Error("The API is disposed");
+        }
+        expect(player.getVideoDuration()).to.eql(
+          videoElement.duration
+        );
+      });
+    });
+
+    describe("getVideoBufferGap", () => {
+      it("should return Infinity in getVideoBufferGap by default", () => {
+        expect(player.getVideoBufferGap()).to.equal(Infinity);
+      });
+    });
+
+    describe("getVideoLoadedTime", () => {
+      it("should return 0 in getVideoLoadedTime by default", () => {
+        expect(player.getVideoLoadedTime()).to.equal(0);
+      });
+    });
+
+    describe("getVideoPlayedTime", () => {
+      it("should return 0 in getVideoPlayedTime by default", () => {
+        expect(player.getVideoPlayedTime()).to.equal(0);
+      });
+    });
+
+    describe("getWallClockTime", () => {
+      it("should return 0 in getWallClockTime by default", () => {
+        expect(player.getWallClockTime()).to.equal(0);
+      });
+    });
+
+    describe("getPosition", () => {
+      it("should return 0 in getPosition by default", () => {
+        expect(player.getPosition()).to.equal(0);
+      });
+    });
+
+    describe("getPlaybackRate", () => {
+      it("should return 1 in getPlaybackRate by default", () => {
+        expect(player.getPlaybackRate()).to.equal(1);
+      });
+    });
+
+    describe("getVolume", () => {
+      it("should return 1 in getVolume by default", () => {
+        expect(player.getVolume()).to.equal(1);
+      });
+    });
+
+    describe("isFullscreen", () => {
+      it("should return false in isFullscreen by default", () => {
+        /* tslint:disable deprecation */
+        expect(player.isFullscreen()).to.equal(false);
+        /* tslint:enable deprecation */
+      });
+    });
+
+    describe("getAvailableVideoBitrates", () => {
+      it("should return [] in getAvailableVideoBitrates by default", () => {
+        expect(player.getAvailableVideoBitrates()).to.eql([]);
+      });
+    });
+
+    describe("getAvailableAudioBitrates", () => {
+      it("should return [] in getAvailableAudioBitrates by default", () => {
+        expect(player.getAvailableAudioBitrates()).to.eql([]);
+      });
+    });
+
+    describe("getVideoBitrate", () => {
+      it("should return undefined in getVideoBitrate by default", () => {
+        expect(player.getVideoBitrate()).to.equal(undefined);
+      });
+    });
+
+    describe("getAudioBitrate", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 167. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=414&issues=AWgOMLyH_MPaBDn0sxTY&open=AWgOMLyH_MPaBDn0sxTY)

[ISSUE_KEY]: AWgOMLyH_MPaBDn0sxTY